src/bsoncxx/private/helpers.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2014 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <bsoncxx/config/prelude.hpp>
+
+#include <bson.h>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view.hpp>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+
+namespace helpers {
+
+BSONCXX_INLINE document::view view_from_bson_t(const bson_t* bson) {
```

Comment:
Even more, you shouldn't apply it as it forces the functions inline, even if the compiler doesn't really want to. We want that for public inlines for ABI reasons, but not for private inlines where we want the compiler to decide as appropriate based on its inlining heuristics.